atomix/cluster/src/test/java/io/atomix/raft/RaftStartupConsistencyCheckTest.java

```diff
@@ -104,7 +111,21 @@ public void shouldHandleRetriedRequestsAfterSnapshotPersist() throws Exception {
         follower.name(),
         new Configurator() {
           @Override
-          public void configure(final MemberId id, final TestSnapshotStore snapshotStore) {
+          public void configure(final MemberId id, final Builder builder) {
+            final var protocol = (TestRaftServerProtocol) builder.protocol;
+            protocol.interceptRequest(
+                InstallRequest.class,
+                r -> {
+                  // only allow the persist method to complete if we know this request was received
+                  // AFTER we already tried persisting (i.e. when persist is currently blocking)
+                  if (snapshotPersistCalled.get()) {
```

Comment:
Hasn't this potential for race conditions 🤔 also wouldn't the above block the raft thread?

Reply:
Ok nevermind I think I get it now, should be fine